### PR TITLE
Use external libs

### LINKED ISSUE #74: Break out device libraries
Break out the device libraries to their own repo (or repos?).

We can then use these libraries for both firmware and apps more easily and keep tags separately from apps.

Will also probably need LGPL licenses to be a little more useful to people.

### PR DESCRIPTION
**Pull request by mchack-work**
Break out device libs to https://github.com/tillitis/tkey-libs

Still needs a rebase on main.

Please have a look and try to build from the instructions in the README.

Closes #74

### CODE REVIEW

**Review by quite — APPROVED** [review]
Looks good. I pushed a fix of some nits.

**Comment by dehanj** [comment]
Tried building locally, works well.

We have to update the toolchain instructions on dev.tillitis, not building with podman since it cannot find the new libs. `make[1]: *** No rule to make target '/src/apps/../../tkey-libs/include/tk1_mem.h', needed by 'signer/main.o'.  Stop.`.
Guess that folder is not mounted to the container and cannot access it.

**Comment by mchack-work** [comment]
Ready for merge.

Rebased on main and squashed. CI passes.

A make target with podman is also included to test with podman.

File: Makefile [hunk 1]
```diff
@@ -36,6 +36,9 @@ reload-rules:
 	udevadm control --reload
 	udevadm trigger
 
+podman:
+	podman run --rm --mount type=bind,source=.,target=/src --mount type=bind,source=../tkey-libs,target=/tkey-libs -w /src -it ghcr.io/tillitis/tkey-builder:1 make -j
```
Comment (dehanj) [inline Makefile:40]: Should be `tkey-builder:2` to reflect the latest container.

Comment (mchack-work) [reply Makefile]: Fixed.

**Comment by mchack-work** [comment]
Added docs on how to use the podman target as well.

**Review by dehanj — COMMENTED** [review]
Minor issue to run podman on mac found, otherwise nothing to comment on.

File: Makefile [hunk 2]
```diff
@@ -36,6 +36,9 @@ reload-rules:
 	udevadm control --reload
 	udevadm trigger
 
+podman:
+	podman run --rm --mount type=bind,source=.,target=/src --mount type=bind,source=../tkey-libs,target=/tkey-libs -w /src -it ghcr.io/tillitis/tkey-builder:2 make -j
```
[the remark this answers is not in the document]
Comment (mchack-work) [reply Makefile:40]: But just including that in the Makefile doesn't work. Is a patch like this working on Mac? If so, please commit it and merge.

```diff
diff --git a/Makefile b/Makefile
index 8b946a3..1e76e51 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,7 @@
 RM=/bin/rm
 
+P := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+
 .PHONY: all
 all: apps tkey-runapp tkey-sign runsign.sh tkey-ssh-agent runtimer runrandom
 
@@ -37,7 +39,7 @@ reload-rules:
 	udevadm trigger
 
 podman:
-	podman run --rm --mount type=bind,source=.,target=/src --mount type=bind,source=../tkey-libs,target=/tkey-libs -w /src -it ghcr.io/tillitis/tkey-builder:2 make -j
+	podman run --rm --mount type=bind,source=$(P),target=/src --mount type=bind,source=$(P)/../tkey-libs,target=/tkey-libs -w /src -it ghcr.io/tillitis/tkey-builder:2 make -j
 
 .PHONY: apps
 apps:
```

[the remark this answers is not in the document]
Comment (mchack-work) [reply Makefile:40]: Or use GNU Make's $(CURDIR). It's perhaps not that important if this works regardless of where the user's working directory really is.

[the remark this answers is not in the document]
Comment (dehanj) [reply Makefile:40]: Not sure if you saw that I edited my comment seconds after my review to use GNU Make's $(CURDIR) - I noticed that my first comment wasn't working. Using CURDIR seem to be the simplest way forward. I'll commit an merge. 
I'm testing on both mac and linux.

**Pull request merged**